sycl/source/handler.cpp

```diff
@@ -417,6 +417,42 @@ void handler::extractArgsAndReqs() {
 
   const bool IsKernelCreatedFromSource = MKernel->isCreatedFromSource();
 
+  // TODO: to do that in compile time.
+  size_t real_KernelArgsNum = UnPreparedArgs.size();
+  for (size_t I = 0; I < UnPreparedArgs.size(); ++I) {
+    const detail::kernel_param_kind_t &Kind = UnPreparedArgs[I].MType;
+    if (Kind == detail::kernel_param_kind_t::kind_accessor) {
+      const int &Size = UnPreparedArgs[I].MSize;
+      // For args kind of accessor Size is information about accessor.
+      // The first 11 bits of Size encodes the accessor target.
+      const access::target AccTarget =
+          static_cast<access::target>(Size & 0x7ff);
+      switch (AccTarget) {
+      case access::target::global_buffer:
+      case access::target::constant_buffer:
+      case access::target::local: {
+        if (!IsKernelCreatedFromSource) {
```

Comment:
Comments would be helpful as to why certain type of args take more space.

Reply:
added a comment